test/e2e/cluster_upgrade_runtimesdk.go

```diff
@@ -239,7 +239,32 @@ func clusterUpgradeWithRuntimeSDKSpec(ctx context.Context, inputGetter func() cl
 		})
 		Expect(err).ToNot(HaveOccurred(), "Lifecycle hook calls were not as expected")
 
+		By("Deleting the workload cluster")
+		cluster := &clusterv1.Cluster{}
+		Eventually(func() error {
+			return input.BootstrapClusterProxy.GetClient().Get(ctx, client.ObjectKey{Namespace: namespace.Name, Name: clusterName}, cluster)
+		}).Should(Succeed())
+
+		// Dump all the logs from the workload cluster before deleting them.
+		input.BootstrapClusterProxy.CollectWorkloadClusterLogs(ctx, cluster.Namespace, cluster.Name, filepath.Join(input.ArtifactFolder, "clusters", cluster.Name))
+
+		Byf("Dumping all the Cluster API resources in the %q namespace", namespace.Name)
+
+		// Dump all Cluster API related resources to artifacts before deleting them.
+		framework.DumpAllResources(ctx, framework.DumpAllResourcesInput{
+			Lister:    input.BootstrapClusterProxy.GetClient(),
+			Namespace: namespace.Name,
+			LogPath:   filepath.Join(input.ArtifactFolder, "clusters", input.BootstrapClusterProxy.GetName(), "resources"),
+		})
+
+		beforeClusterDeleteHandler(ctx, input.BootstrapClusterProxy.GetClient(), namespace.Name, clusterName, input.E2EConfig.GetIntervals(specName, "wait-machine-upgrade"))
+
+		framework.DeleteClusterAndWait(ctx, framework.DeleteClusterAndWaitInput{
```

Comment:
Currently not passing - due to (I think) both this and the beforeClusterDeleteHandler blocking

Reply:
Yuvaraj's comment + Yup. 

I think you have to call `DeleteCluster` and `WaitForClusterDeleted` separately, but this should be fine

Reply:
Probably fine to ignore `GetCAPIResources` as it's called in AfterEach again